File: PUNTOS DE VENTA/puntoDeVenta/puntoDeVenta/Classes/UIReportAddon1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;
using System.Windows.Forms;
using System.Data;
using puntoDeVenta.Forms;
namespace puntoDeVenta.Classes
{
    class UIReportAddon1 : IGraphicalAddon
    {
        private DataSet _dataSource;
        private Form _parentForm;
        private Object _saycSetting;
        public DataSet DataSource {
            set {
                this._dataSource = value;
            }
        }
        public String Name {
            get {
                return "PVenta";
            }
        }
        public String NameText{
            get {
                return "Punto de Venta";
            }
        }
        public String idProduct{
            get {
                return "B101";
            }
        }
        public String Category {
            get{
                return "Ventas";
            }
        }
        public Boolean mdiStatus{
            get{
                return false;
            }
        }
        public Image Image{
            get{
                return null;
            }
        }
        public Form ParentForm{
            set{
                this._parentForm = value;
            }
        }
        public Object SaycSetting{
            set{
                this._saycSetting = value;
            }
        }
        public void onClick(Object sender, System.EventArgs e){
            PuntoVenta newUiReportingAddonForm = new PuntoVenta(this._saycSetting);
            if(this.mdiStatus){
                newUiReportingAddonForm.MdiParent = this._parentForm;
                newUiReportingAddonForm.MaximizeBox = true;
            }
            newUiReportingAddonForm.ShowDialog();
        }   
    }
}

File: Eximiciones Y Vacaciones/FormPrueba/Classes/UIReportAddon1.vb
Public Class UIReportAddon1
    Implements IGraphicalAddon
    Private _dataSource As DataSet
    Private _parentForm As Form
    Private _saycSetting
    Public WriteOnly Property DataSource As System.Data.DataSet Implements IGraphicalAddon.DataSource
        Set(ByVal value As System.Data.DataSet)
            _dataSource = value
        End Set
    End Property
    Public ReadOnly Property Name As String Implements IGraphicalAddon.Name
        Get
            Return "EximicionesYVacaciones"
        End Get
    End Property
    Public ReadOnly Property NameText As String Implements IGraphicalAddon.NameText
        Get
            Return "Eximiciones y Vacaciones"
        End Get
    End Property
    Public ReadOnly Property idProduct As String Implements IGraphicalAddon.idProduct
        Get
            Return "A109"
        End Get
    End Property
    Public ReadOnly Property Category As String Implements IGraphicalAddon.Category
        Get
            Return "Operaciones"
        End Get
    End Property
    Public ReadOnly Property mdiStatus As Boolean Implements IGraphicalAddon.mdiStatus
        Get
            Return False
        End Get
    End Property
    Public ReadOnly Property Image As Image Implements IGraphicalAddon.Image
        Get
            Return My.Resources._7days
        End Get
    End Property
   
    Public WriteOnly Property ParentForm As Form Implements IGraphicalAddon.ParentForm
        Set(ByVal ParentForm As Form)
            Me._parentForm = ParentForm
        End Set
    End Property
    Public WriteOnly Property SaycSetting As Object Implements IGraphicalAddon.SaycSetting
        Set(ByVal setting As Object)
            Me._saycSetting = setting
        End Set
    End Property
    Public Sub OnClick(ByVal sender As Object, ByVal e As System.EventArgs) Implements IGraphicalAddon.OnClick
        Dim newUiReportingAddonForm As New Eximiciones(_saycSetting)
        If mdiStatus Then
            newUiReportingAddonForm.MdiParent = Me._parentForm
            newUiReportingAddonForm.MaximizeBox = True
        End If
        newUiReportingAddonForm.ShowDialog()
    End Sub
End Class
